### PR TITLE
Fix block schema recognition for Plugin Framework

### PR DESCRIPTION
**Pull request by t0yv0**
A few more tests and fixes for recognizing schema of Plugin Framework blocks.

### CODE REVIEW

File: pf/internal/schemashim/convert_type.go [hunk 1]
```diff
@@ -46,8 +46,10 @@ func convertType(typ pfattr.Type) (shim.ValueType, error) {
 		return shim.TypeInvalid, nil
 	default:
 		switch tftype.(type) {
-		case tftypes.List, tftypes.Set:
```
Comment (t0yv0) [inline pf/internal/schemashim/convert_type.go:49]: This was a bug, set-nested blocks were incorrectly recognized as lists.

**Review by iwahbe — APPROVED** [review]
Thanks for all the additional tests.

File: pkg/tfshim/schema/schema.go [hunk 2]
```diff
@@ -168,3 +168,85 @@ func (m SchemaMap) Set(key string, value shim.Schema) {
 func (m SchemaMap) Delete(key string) {
 	delete(m, key)
 }
+
+func FromSchema(s shim.Schema) *Schema {
```
Comment (iwahbe) [inline pkg/tfshim/schema/schema.go:172]: Where are the additions to this file used?

Comment (t0yv0) [reply pkg/tfshim/schema/schema.go:172]: Oops I meant to remove it. I didn't end up using this. Was thinking it can help writing test.

**Pull request merged**